### ISSUE TITLE
Device info leak to Worker

### ISSUE THREAD

**Issue by antarcticoil** [polcak/jsrestrictor#199]
Cpu cores, ram, gpu info is leaked above fingerprinting test. It's masked to Navigator, but not to Worker. I modified all wrappers to strict, but it was no use either.

Pages affected: https://abrahamjuliot.github.io/creepjs/
JShelter Version: 0.11.4
OS: Windows 10 x64
Browser: Vivaldi 5.7.2921.65 and Opera 96.0.4693.80. They are all chromium 110. I only checked these two browsers, but I think other chromium based browsers without embedded anti-fingerprinting have the same problem.

Vivaldi also sends the device memory info by header, so it's not masked even to Navigator. I'm not sure if it yields the same result regardless of other factors like OS. I know this is not a jshelter's issue, but inform to you for your easy understanding. I really appreciate for all your hard work.

**Comment by TechnologyClassroom** [comment]
I am having trouble reproducing this issue. The test never completes for me. It just displays "FP ID: Computing..." indefinitely.

**Comment by antarcticoil** [comment]
> I am having trouble reproducing this issue. The test never completes for me. It just displays "FP ID: Computing..." indefinitely.

Can you tell me your environment? I tested it on Edge and got the same result. I waited about 5 seconds on Edge.

**Comment by TechnologyClassroom** [comment]
Browser: Abrowser 111.0.1 (64-bit) (Fork of Firefox)
JShelter Version: 0.11.4
OS: Trisquel 9 (Fork of Ubuntu)

**Comment by antarcticoil** [comment]
Yes, Firefox stuck on "computing" with jshelter. In the first place, I think Firefox does not leak real device info to Worker even without jshelter. This issue seems to be limited to Chromium.

**Comment by polcak** [comment]
In essence, this is a duplicate of https://pagure.io/JShelter/webextension/issue/80. In case you have further information, please, let us know.

At the moment, we are working on migration to Manifest v3. I hope that this issue will be solved during the migtation. I think that a fix for Manifest v2 version is unlikely.
